matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/GetEventTask.kt

```diff
@@ -16,28 +16,49 @@
 
 package org.matrix.android.sdk.internal.session.room.timeline
 
+import org.matrix.android.sdk.api.extensions.tryOrNull
 import org.matrix.android.sdk.api.session.events.model.Event
+import org.matrix.android.sdk.internal.crypto.EventDecryptor
+import org.matrix.android.sdk.internal.crypto.algorithms.olm.OlmDecryptionResult
 import org.matrix.android.sdk.internal.network.GlobalErrorReceiver
 import org.matrix.android.sdk.internal.network.executeRequest
 import org.matrix.android.sdk.internal.session.room.RoomAPI
 import org.matrix.android.sdk.internal.task.Task
 import javax.inject.Inject
 
-// TODO Add parent task
-
-internal class GetEventTask @Inject constructor(
-        private val roomAPI: RoomAPI,
-        private val globalErrorReceiver: GlobalErrorReceiver
-) : Task<GetEventTask.Params, Event> {
-
-    internal data class Params(
+internal interface GetEventTask : Task<GetEventTask.Params, Event> {
+    data class Params(
             val roomId: String,
-            val eventId: String
+            val eventId: String,
     )
+}
 
-    override suspend fun execute(params: Params): Event {
-        return executeRequest(globalErrorReceiver) {
+internal class DefaultGetEventTask @Inject constructor(
+        private val roomAPI: RoomAPI,
+        private val globalErrorReceiver: GlobalErrorReceiver,
+        private val eventDecryptor: EventDecryptor
+) : GetEventTask {
+
+    override suspend fun execute(params: GetEventTask.Params): Event {
+        val event = executeRequest<Event>(globalErrorReceiver) {
             apiCall = roomAPI.getEvent(params.roomId, params.eventId)
         }
+
+        // Try to decrypt the Event
+        if (event.isEncrypted()) {
+            tryOrNull(message = "Unable to decrypt the event") {
+                eventDecryptor.decryptEvent(event, "")
+            }
+                    ?.let { result ->
+                        event.mxDecryptionResult = OlmDecryptionResult(
```

Comment:
Maybe we should do that directly from the eventDecryptor?

Reply:
This is out of scope of this PR